models/issue.go

```diff
@@ -83,7 +84,8 @@ func (issue *Issue) loadTotalTimes(e Engine) (err error) {
 
 // IsOverdue checks if the issue is overdue
 func (issue *Issue) IsOverdue() bool {
-	return util.TimeStampNow() >= issue.DeadlineUnix
+	deadline := issue.DeadlineUnix.AsTime().UTC().Round(24 * time.Hour)
+	return time.Now().UTC().After(deadline)
```

Comment:
I'd rather change the behaviour when saving a due date to save it on 23:59:59 instead of 00:00 rather than changing this when showing the due date.

Reply:
```Go
util.TimeStampNow() >= issue.DeadlineUnix + util.TimeStamp(time.Hour * 24 / time.Second)
```

Reply:
I agree with kolaente

Reply:
Ok thinking about the solution It would be like this

```go
if form.Deadline != nil && !form.Deadline.IsZero() {
	t := form.Deadline
	deadLine := time.Date(t.Year(), t.Month(), t.Day(), 23, 59, 59, 0, t.Location())
	form.Deadline = &deadLine
	deadlineUnix = util.TimeStamp(form.Deadline.Unix())
}
```
This is an example for the issue deadline case. What do you think @lunny, @kolaente, @Bwko ? Maybe this code could be simplified.

Reply:
I'd only set it to 23:59:59 if it is set to 00:00:00. Maybe someone wants to use it in some kind of frontend with exact deadlines. Or maybe don't modify it at all on the backend and only in gitea's frontend?

Reply:
@rvillablanca I don't against @kolaente 's idea, but that you have to consider old data migration.

Reply:
@kolaente I don't think the frontend currently supports ___exact deadlines___.

Reply:
@adelowo Ours doesn't, but the api does.

Reply:
I don't think we should support exact deadlines (with time)